### ISSUE TITLE
Site.rb provider action [:add] doesn't set a specified custom log_directory attribute.

### ISSUE THREAD

**Issue by dschrander** [opscode-cookbooks/iis#159]
Specifying  a log_directory attribute is ignored when adding a new site using the [:add, :start] options for the iis_site provider.
The same attribute gets set when using the [:config] action for the iis_site provider.

As a workaround I have created the site and ran a [:config] right after that, just setting the custom log_directory.

Can this be fixed in the add part of the iis_site provider please?

**Comment by EasyAsABC123** [comment]
You are correct, I'll fix this tonight.  Basically the :config operation needs to be turned into a function and called from the :add operation.

Justin Schuhmann

**Comment by dschrander** [comment]
Thanks for the quick response. Looking forward to it!

Cheers,
David Schrander

**Comment by EasyAsABC123** [comment]
Never a problem, I'm running wires to move my office so I don't have a computer this second.  Hopefully I'll have it done today.

**Comment by EasyAsABC123** [comment]
@dschrander please test the bug-site_add_does_note_config branch found here https://github.com/opscode-cookbooks/iis/tree/bug-site_add_does_not_config

**Comment by dschrander** [comment]
Hi Justin,
Thanks for the update. It does configure the log_directory now, but unfortunately I get a 'NoMethodError' error when also using 'bindings' (undefined method `bindings=' for Chef::Resource::IisSite).

It does however create the site and it's bindings properly and starts it, but every chef run fails on above error.
When reverting back to the master branch, the error is gone (no logging configured obviously).

Cheers,
David

**Comment by EasyAsABC123** [comment]
Sounds good, I'll get that resolved or inform you of what I think it is, if you could post the real error that would be helpful, real as in full

**Comment by dschrander** [comment]
Sure, here's what I got in my stacktrace output:
NoMethodError: iis_site[Test Site] (test_iis_role::default line 64) had an error: NoMethodError: undefined method `bindings=' for Chef::Resource::IisSite

**Comment by EasyAsABC123** [comment]
Can you post the iis_site provider declaration from your recipe?

**Comment by dschrander** [comment]
Compiled Resource:
 ------------------
 # Declared in C:/chef/cache/cookbooks/test_iis_role/recipes/default.rb:64:in `block in from_file'

 iis_site("Test Site") do
   action [:add, :start]
   retries 0
   retry_delay 2
   guard_interpreter :default
   cookbook_name "test_iis_role"
   recipe_name "default"
   protocol :http
   port 80
   application_pool "Test Site"
   path "D:\\AppSites/Test/Public\\www"
   bindings "http/*:80:www.test.com,http/*:80:www.test2.com"
   site_name "Test Site"
 end

**Comment by EasyAsABC123** [comment]
That's all I need next time I am at a computer I'll resolve this

**Comment by EasyAsABC123** [comment]
not seeing any bugs, can you verify that you downloaded https://github.com/opscode-cookbooks/iis/archive/bug-site_add_does_not_config.zip and uploaded it?

**Comment by dschrander** [comment]
Yes, I have. But even if I throw away my local repo, download the master from git it breaks on this exact same thing.
I have noticed that in the last commit '07e6e34cd3efb5759466309c494af8ea6cebe55e' that ksubrama made 5 days ago is "Showing  45 changed files  with 343 additions and 304 deletions."

This was done without updating the version and now doesn't work for me anymore. Like I said, I had to get it from the supermarket to get it working again.

So it is nog the branch with the fix, that is probably just fine. It's what's been commited to master that broke it for me.
